pom.xml

```diff
@@ -19,28 +19,158 @@
   <properties>
     <github.repository>IDEasy</github.repository>
     <ide_version>${revision}</ide_version>
-    <java.version>17</java.version>
+    <java.version>21</java.version>
+    <maven.compiler.source>21</maven.compiler.source>
+    <maven.compiler.target>21</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <jline.version>3.24.1</jline.version>
+    <jansi.version>2.4.1</jansi.version>
+    <jackson.version>2.18.2</jackson.version>
+    <xmlunit.version>2.10.0</xmlunit.version>
+    <wiremock.version>3.11.0</wiremock.version>
+    <javafx.version>21</javafx.version>
+    <testfx.version>4.0.18</testfx.version>
   </properties>
 
-  <dependencies>
-    <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
-      <version>5.12.0</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.assertj</groupId>
-      <artifactId>assertj-core</artifactId>
-      <version>3.24.2</version>
-      <scope>test</scope>
-    </dependency>
-  </dependencies>
```

Comment:
why did you remove this?
All our 3 child modules are Java projects that obviously have tests and since our test infrastructure is always JUnit + AssertJ, you now have to add the dependencies redundantly in all child modules.
You could have also just removed the version here.

However, you did nothing wrong and having all real dependencies in the child POMs is also fine to make things explicit.
Therefore, I consider this nice-to-have or personal flavour and simply resolve it directly.